packages/nvidia_nat_nemo_customizer/src/nat/plugins/customizer/dpo/trainer_adapter.py

```
@@ -400,13 +400,51 @@ async def wait_until_complete(self, ref: TrainingJobRef, poll_interval: float |
+
+                    # If deployment was expected, raise an error
+                    if self.adapter_config.deploy_on_completion:
+                        raise RuntimeError(f"Training job {ref.run_id} was canceled at {progress_str}: {error_msg}. "
+                                           f"Model was not trained and will NOT be deployed. "
+                                           f"Evaluation will fail because the model does not exist. "
+                                           f"\n\nACTION REQUIRED:"
+                                           f"\n1. Check if the job was manually canceled or timed out"
+                                           f"\n2. Review NeMo MS platform health and resource availability"
+                                           f"\n3. Consider increasing deployment_timeout_seconds in config"
+                                           f"\n4. Use a fresh namespace to avoid conflicts: namespace: nat-dpo-test-v2"
+                                           f"\n5. Retry training: nat finetune --config_file=...")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)